lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc

```diff
@@ -143,3 +143,26 @@ function dosomething_reportback_filter_flagged_reportbacks($params) {
   return $params;
 }
 
+/**
+ * Filter out requested Reportbacks based on associated Signup. 
+ *
+ * @param string $run_id Campaign run id. 
+ * @param string $sid Signup id. 
+ * @return mixed
+ */
+function dosomething_reportback_filter_reportbacks_by_signups($run_id, $sid) {
+  $reportbacks = db_query("SELECT DISTINCT s.sid, r.rbid, r.run_nid, s.run_nid
+                          FROM dosomething_signup s 
+                          INNER JOIN dosomething_reportback r on s.run_nid = r.run_nid
+                          WHERE s.uid = r.uid;");
```

Comment:
So this feels like its on the right track, but you could likely do it all in the `dosomething_signup.query.inc` file, so it's all in one hit to the DB. Just add it to the query there with a join on the conditions :) Then in the **SignupTransformer.php** you can do  `$this->reportbacks = Reportback::get($rbid)` without the try catch cause it should return just one :)